### PR TITLE
Form fixes

### PR DESCRIPTION
**Pull request by nyalldawson**
Fixes some various issues with attribute table dialog.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Would appreciate a review here - attribute form changes scare me, they always seem to have unwanted side-effects...!

File: src/gui/attributetable/qgsdualview.cpp [hunk 1]
```diff
@@ -912,9 +916,11 @@ void QgsDualView::setFeatureSelectionManager( QgsIFeatureSelectionManager *featu
 void QgsDualView::setAttributeTableConfig( const QgsAttributeTableConfig &config )
 {
   mConfig = config;
-  mLayer->setAttributeTableConfig( config );
-  mFilterModel->setAttributeTableConfig( config );
-  mTableView->setAttributeTableConfig( config );
+  if ( mLayer )
```
Comment (m-kuhn) [inline src/gui/attributetable/qgsdualview.cpp]: Do we need this check?
It only guards the first access to layer. I guess if there is never a crash with line 921 we can also safely assume that layer is present for 920.

**Comment by m-kuhn** [comment]
> attribute form changes scare me, they always seem to have unwanted side-effects...!

True story!
Looks good to me as is, but it's even harder to predict side-effects by just reading the code... So I'd say best to merge early in a dev cycle.

File: src/gui/attributetable/qgsdualview.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/gui/attributetable/qgsdualview.cpp]: Good point. Coverity would not be happy!

**Comment by nyalldawson** [comment]
Thanks @m-kuhn!

**Pull request merged**